gpt4all-api/gpt4all_api/app/api_v1/routes/chat.py

```diff
@@ -38,38 +49,115 @@ class ChatCompletionResponse(BaseModel):
     choices: List[ChatCompletionChoice]
     usage: ChatCompletionUsage
 
+
+class CompletionStreamResponse(BaseModel):
+    id: str
+    object: str = 'text_completion'
+    created: int
+    model: str
+    choices: List[ChatCompletionChoice]
+
 router = APIRouter(prefix="/chat", tags=["Completions Endpoints"])
 
+# Add a function to save the chat session
+def save_chat_session(session_id, data):
+    os.makedirs('chat_sessions', exist_ok=True)
+    with open(f'chat_sessions/{session_id}.json', 'w') as file:
+        json.dump(data, file)
+
+def load_chat_session(session_id):
+    try:
+        with open(f'chat_sessions/{session_id}.json', 'r') as file:
+            content = file.read()
+            if not content:
+                logger.error(f"Empty or invalid content in session file: {session_id}")
+                return []  # Return an empty list if file is empty or invalid
+            return json.loads(content)
+    except FileNotFoundError:
+        logger.error(f"File not found for session: {session_id}")
+        return []  # Return an empty list if file not found
+    except json.JSONDecodeError as e:
+        logger.error(f"JSON decode error for session {session_id}: {str(e)}")
+        return []  # Return an empty list if JSON is invalid
+
+
+@router.get("/completions/final")
+async def final_completion(session_id: str):
+    '''
+    Return final conversation answers
+    '''
+    chat_history = load_chat_session(session_id)
+    return chat_history
+
+# Initialize GPT4All model outside of the endpoint function
+gpt_model = GPT4All(model_name=settings.model, model_path=settings.gpt4all_path)
+print("MODEL: "+settings.model)
```

Comment:
This diverges away from the OpenAI openapi implementation. The goal of this library is to maintain an interface to GPT4All models but with an openai openapi compatible schema.